src/plugins/remove/RemoveAction.js

```diff
@@ -132,4 +139,29 @@ export default class RemoveAction {
             && parentType.definition.creatable
             && Array.isArray(parent.composition);
     }
+
+    activeTransaction() {
+        return this.openmct.objects.getActiveTransaction();
+    }
+
+    startTransaction() {
+        if (!this.openmct.editor.isEditing()) {
+            this.openmct.objects.startTransaction();
+        }
+    }
+
+    saveTransaction() {
+        const transaction = this.activeTransaction();
+
+        if (!transaction || this.openmct.editor.isEditing()) {
```

Comment:
Again, we need to separate edit mode from transactions. The remove action doesn't really care about whether edit mode is active, what it cares about is whether _it owns the active transaction_. If it doesn't own the current transaction, it shouldn't commit it. It knows whether or not it owns the transaction based on whether or not it created it in `startTransaction()`. I'd suggest that perhaps `startTransaction` should save the returned transaction as a property. When when we get into `saveTransaction` if `this.#transaction` exists, then it has a transaction to commit in this step.

Reply:
We may want to make note of this in other places in the code, I know it's in notebooks as well.

Reply:
Ok, so this made me think of a situation where you're editing something (notebook, display layout) then you remove an item from the tree, then you cancel out of the edit you were making, we'd lose the remove transaction at that point. It's a bit edgy (case-wise ;) ), but it's a valid case we should consider.

Reply:
Agreed. Edit mode is an application-wide state, but our current UI suggests it's restricted to the navigated object, so I think there is an inconsistency between user expectation and our actual behavior which we should address...one day.